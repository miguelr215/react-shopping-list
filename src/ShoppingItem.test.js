import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShoppingItem from './ShoppingItem.js';

describe('ShoppingItem Component', () => {
    it('renders empty given no item', () => {
        const wrapper = shallow(<ShoppingItem />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the item when supplied', () => {
        // We're adding props to the ShoppingItem for different test cases
        const testItem = {name:'test-item', checked: false}
        const wrapper = shallow(<ShoppingItem item={testItem} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('strikes through checked items', () => {
        const testItem = {name:'test-item', checked: true}
        const wrapper = shallow(<ShoppingItem item={testItem} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})