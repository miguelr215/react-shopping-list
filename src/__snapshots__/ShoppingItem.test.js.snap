// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingItem Component renders empty given no item 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": null,
      }
    }
  />
  <button
    onClick={[Function]}
    type="button"
  >
    check
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
</li>
`;

exports[`ShoppingItem Component renders the item when supplied 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": null,
      }
    }
  >
    test-item
  </h2>
  <button
    onClick={[Function]}
    type="button"
  >
    check
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
</li>
`;

exports[`ShoppingItem Component strikes through checked items 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": "line-through",
      }
    }
  >
    test-item
  </h2>
  <button
    onClick={[Function]}
    type="button"
  >
    check
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
</li>
`;
